internal/resolver/delegatingresolver/delegatingresolver.go

```diff
@@ -217,25 +238,35 @@ func (r *delegatingResolver) updateClientConnStateLocked() error {
 		proxyAddr = resolver.Address{Addr: r.proxyURL.Host}
 	}
 	var addresses []resolver.Address
-	for _, targetAddr := range (*r.targetResolverState).Addresses {
+	for _, targetAddr := range curState.Addresses {
```

Comment:
nit: `curState` fells a bit unclear. The reader doesn't know if it's the proxy resolver's state or the targetResolver's without looking up the definition. We should name it `targetResolverState` or something similar.